dpmData.py

```diff
@@ -1,47 +1,83 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
-Created on Mon Apr 20 11:43:17 2020
+Created on Tue Jun 16 13:20:08 2020
 
 @author: snag
 """
+import argparse
+import datetime
+import sys
 import pandas as pd
 import DPM
 
-DEVICE_LIMIT = 0
-DEVICE_LIST = []
+def main():
+    parser = argparse.ArgumentParser()
+    #Add positional/optional parameters
+    parser.add_argument('-d', '--device_limit', help='Device Limit', type=int, default=0)
+    parser.add_argument('-f', '--device_file', help='device file name')
+    parser.add_argument('-o', '--output_file', help='output file name', default='data.h5')
+    parser.add_argument('-n', '--node', help='node name', default=None, type=str)
+        #Error when not using the default value
+    #group 1
+    parser.add_argument("-s", "--start_date", type=lambda s: datetime.datetime.strptime(s, '%Y-%m-%d'), help='Enter the start time/date. Do not use the duration tag', required=False)
+    parser.add_argument("-e", "--end_date", type=lambda s: datetime.datetime.strptime(s, '%Y-%m-%d'), help='Enter the start time/date. Do not use the duration tag', required=False)
+    #group 2
+    parser.add_argument("-du", "--duration", help="Enter LOGGERDURATION in sec", required=False)
+    #Parse the args
+    args = parser.parse_args()
+    sys.stdout.write(str(hdf_code(args)))
 
-with open('DeviceList.csv') as f:
-    DEVICE_LIST = [line.rstrip() for index, line in enumerate(f)
-                   if index < DEVICE_LIMIT or DEVICE_LIMIT < 1]
+def hdf_code(args):
+    START_DATE = args.start_date
+    END_DATE = args.end_date
+    DURATION = args.duration
+    DEVICE_LIMIT = args.device_limit
+    DEVICE_FILE = args.device_file
+    OUTPUT_FILE = args.output_file
+    #NODE = args.node #unused
 
-dpm = DPM.Blocking()
-for index, device in enumerate(DEVICE_LIST):
-    dpm.add_entry(index, device)
+    if DURATION and (START_DATE or END_DATE):
+        print("-d and -s|-e are mutually exclusive! Exiting ...")
+        sys.exit(2)
+    elif(START_DATE and END_DATE):
+        DURATION = END_DATE - START_DATE
+        #print(int((END_DATE - START_DATE).total_seconds()))
+        DURATION = str(DURATION)
+        DURATION = 'LOGGERDURATION:' + DURATION
```

Comment:
Duration is distinct from start and end. See, https://cdcvs.fnal.gov/redmine/projects/acsys-dpm/wiki/Data_Sources#LOGGERT1T2NODE.

In fact, this isn't doing what it says it's doing because the DPM data source is duration and the end time is ALWAYS now even though we set an end date.

Reply:
For duration, if I do now - START_DATE instead, and now prints in '%Y-%m-%d %H:%M:%S', should I include %H:%M:%S for the START_DATE user input too? That only takes %Y-%m-%d currently.

Reply:
For this conditional block where `START_DATE` and `END_DATE` are used, the word `DURATION` should not be used. We want to pass the start and end using the t1:t2 notation, https://cdcvs.fnal.gov/redmine/projects/acsys-dpm/wiki/Data_Sources#LOGGERT1T2NODE.
I would call it something like `TIME_WINDOW` and do `TIME_WINDOW = 'LOGGER:' + t1 + ':' + t2`.
Or even slimmer, `TIME_WINDOW = f'LOGGER:{t1}:{t2}'`

Let me know if this doesn't make sense and I can submit a modification. I think.

Reply:
I've made some changes but would need clarification to make sure I'm understanding this correctly. Can we schedule a call for today?